### PR TITLE
Added nodes/pipes to excluded to the skel proc

### PR DESCRIPTION
**Pull request by frank-hulo**
Provide a summary of the proposed changes, describe tests and documentation, and review the acknowledgement below. 
#347 
 
## Summary
Added nodes and links to exclude in the skelenazation process. It is handled in the same manner as the nodes and links with controls. 

## Tests and documentation
Test copied from nodes and links with controls, should have the same impact. 

Two new inputs added to the initialization of the class, so they can be extended to the nodes -links with controls. They could also be separated, but the if statements and the three separate stages are fairly complex, so I didn't want to add another `and` to it. 

Only the naming could now be improved in the if statements, instead nodes_with_controls -> nodes_to_exclude, which makes it easier to read. 
 
## Acknowledgement
By contributing to this software project, I acknowledge that I have reviewed the [software quality assurance guidelines](https://wntr.readthedocs.io/en/stable/developers.html) and that my contributions are submitted under the [Revised BSD License](https://wntr.readthedocs.io/en/stable/license.html).

### CODE REVIEW

**Comment by kaklise** [comment]
Thanks for adding this feature!  Since tanks and reservoirs are not included in skeletonization, I suggest changing the term node to junction in the updated code and documentation.  I'll make some notes in the code review.  Also, I agree that `junc_with_controls` and `pipe_with_controls` should be renamed to` juncs_to_exclude`, `pipes_to_exclude` in the skeletonization method.

File: wntr/tests/test_morph.py [hunk 1]
```diff
@@ -315,6 +315,29 @@ def test_skeletonize_with_controls(self):
...
+        link.diameter = 16 * 0.0254
+        link = wn.get_link("11")
+        link.diameter = 16 * 0.0254
+
+        skel_wn = wntr.morph.skeletonize(wn, 12.0 * 0.0254, use_epanet=False)
+
+        self.assertEqual(skel_wn.num_nodes, wn.num_nodes - 17)
+        self.assertEqual(skel_wn.num_links, wn.num_links - 22)
+
     def test_series_merge_properties(self):
```
Comment (kaklise) [inline wntr/tests/test_morph.py:341]: This test would be stronger by asserting that pipe 60 and junction 13 are in skel_wn.

**Review by kaklise — CHANGES_REQUESTED** [review]

File: wntr/tests/test_morph.py (hunk 1 above)
Comment (kaklise) [reply wntr/tests/test_morph.py:341]: @frank-hulo Will you be able to address the review comments?  If not, we can make the minor updates in house.

Comment (frank-hulo) [reply wntr/tests/test_morph.py:341]: @kaklise I could finish it within the next two weeks.

Comment (kaklise) [reply wntr/tests/test_morph.py:341]: @frank-hulo We'd like to get this PR merged into main.  Let me know if you can address the comments.  I can finish it up next week if needed.

**Pull request merged**